spec/tapioca/compilers/_symbol_table_compiler_spec.rb

```diff
@@ -1703,5 +1703,120 @@ def foo(*_, &_); end
         RUBY
       )
     end
+
+    it("compiles signatures and structs in source files") do
+      expect(
+        compile(template(<<~RUBY))
+          class Foo
+            extend(T::Sig)
+
+            sig { params(a: Integer, b: String).void }
+            def foo(a, b:)
+            end
+
+            sig { params(a: Integer, b: String).returns(Integer) }
+            def bar(a, b:)
+            end
+
+            sig { type_parameters(:U).params(a: T.type_parameter(:U)).returns(T.type_parameter(:U)) }
+            def baz(a)
+              a
+            end
+
+            class << self
+              extend(T::Sig)
+
+              sig { void }
+              def quux
+              end
+            end
+          end
+
+          class Bar < T::Struct
+            const :foo, Integer
+            prop :bar, String
+            const :baz, T::Hash[String, T.untyped]
+            prop :quux, T.untyped, default: [1, 2, 3]
+          end
+
+          class Baz
+            extend(T::Sig)
+            extend(T::Helpers)
+
+            abstract!
+
+            sig { abstract.void }
+            def do_it
+            end
+          end
+
+          module Quux
+            extend(T::Sig)
+            extend(T::Helpers)
+
+            interface!
+
+            sig { abstract.returns(Integer) }
+            def something
+            end
+
+            class Concrete
+              extend(T::Sig)
+              include Quux
+
+              sig { override.returns(Integer) }
+              def something
+              end
+            end
+          end
+        RUBY
+      ).to(
+        eq(template(<<~RUBY))
+          class Bar < ::T::Struct
+            const :foo, Integer
+            prop :bar, String
+            const :baz, T::Hash[String, T.untyped]
+            prop :quux, T.untyped
+
+            def self.inherited(s); end
+          end
+
+          class Baz
+            abstract!
+
+            def initialize(*args, &blk); end
+
+            sig { abstract.void }
+            def do_it; end
+          end
+
+          class Foo
+            sig { params(a: Integer, b: String).returns(Integer) }
+            def bar(a, b:); end
+            sig { type_parameters(:U).params(a: T.type_parameter(:U)).returns(T.type_parameter(:U)) }
```

Comment:
It would be easier to read if we added a blank line between definitions when there is a signature.

Reply:
Ugh, this is proving to be non-trivial since we are building strings just-in-time. I think I will leave it like this for now and revisit later.